debug/antiflood/debugger.go

```diff
@@ -0,0 +1,179 @@
+package antiflood
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/debug"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/storage/lrucache"
+)
+
+const sizeUint32 = 4
+const sizeUint64 = 8
+const sizeBool = 1
+const newLineChar = "\n"
+const minIntervalInSeconds = 1
+
+var log = logger.GetOrCreate("debug/antiflood")
+
+type event struct {
+	pid           core.PeerID
+	topic         string
+	numRejected   uint32
+	sizeRejected  uint64
+	isBlackListed bool
+}
+
+// Size returns the size of an event instance
+func (ev *event) Size() int {
+	return len(ev.pid) + len(ev.topic) + sizeUint32 + sizeUint64 + sizeBool
+}
+
+func (ev *event) String() string {
+	return fmt.Sprintf("pid: %s, topic: %s, num rejected: %d, size rejected: %d, is blacklisted: %v",
+		ev.pid.Pretty(), ev.topic, ev.numRejected, ev.sizeRejected, ev.isBlackListed)
+}
+
+type debugger struct {
+	mutCriticalArea   sync.RWMutex
+	cache             storage.Cacher
+	intervalAutoPrint time.Duration
+	printEventHandler func(data string)
```

Comment:
Naming a bit unfortunate (because we use the suffix `handler` with other meaning usually) - could be `printEventFunc`, `eventPrinter`.

Reply:
printEventFunc shall be